### ISSUE TITLE
Application load balancer

### ISSUE THREAD

**Issue by miksir** [yandex-cloud/terraform-provider-yandex#183]
1. Can't add certificate to backend group
Error: http_backend.3749456530.tls.0.validation_context: '': source data must be an array or slice, got struct
```
    tls {
      validation_context {
          trusted_ca_bytes = data.terraform_remote_state.es.outputs.ca_ssl
      }
    }
  }
```
Same error in case of import configuration

In case of create ALB from scratch - there is no error but certificate missed

2.  Can't import yandex_alb_virtual_host because there is no ID of virtual host.

**Comment by apilikov** [comment]
Thanks for reaching out. Could you please provide minimal reproducing example so we can better understand what's working and what's not. Thank you!

**Comment by miksir** [comment]
1. Just need to create backend group with backend with HTTPS protocol and add PEM-encoded certificate
`terraform import --allow-missing-config yandex_alb_backend_group.test xxxx`
Error: http_backend.NNNNNNN.tls.0.validation_context: '': source data must be an array or slice, got struct

2. Cant import yandex_alb_virtual_host because unknown id
Example: `yc application-load-balancer virtual-host --http-router-id=xxx list`
```
+----------+-------------+-------------+
|   NAME   | AUTHORITIES | ROUTE COUNT |
+----------+-------------+-------------+
| rabbitmq |             |           1 |
+----------+-------------+-------------+
```
There is no ID. Same as in Web GUI.

**Comment by apilikov** [comment]
We'll take a look closely at PEM certificate issue, thank you.
To address import problem use ID = http_router_id/vhost_name in your import statement, e.g.
`terraform import yandex_alb_virtual_host.translate_v1_l7_vh a5dn4bvq4ltqvm9kguha/translate_api_v1_host`

**Comment by miksir** [comment]
http_router_id/vhost_name works fine, thanks. Probably need to fix documentation.

**Comment by miksir** [comment]
One more issue with yandex_alb_virtual_host in http_route_action.upgrade_types 
In GUI "WebSocket" is checkbox. 
<img width="573" alt="Screenshot 2021-08-19 at 15 12 51" src="https://user-images.githubusercontent.com/2039824/130066464-f31ca396-aa9e-4310-ba44-a5a8c839558f.png">
But it is not imported and if I specify "websocket" or "WebSocket" in http_route_action.upgrade_types  - it's not turn on checkbox (example on image)

**Comment by apilikov** [comment]
> http_router_id/vhost_name works fine, thanks. Probably need to fix documentation.

We did fix it. Thank you!

**Comment by miksir** [comment]
And one more issue in yandex_alb_load_balancer
```
  listener {
    name = "test"
    endpoint {
      ports = [ 443 ]
      address {
        internal_ipv4_address {
          subnet_id = local.networks[0]
        }
      }
    }

  ...
  }
```
works fine but changes are detected every "terrafom apply"
```
      ~ listener {
            name = "test"

          ~ endpoint {
                # (1 unchanged attribute hidden)

              ~ address {

                  ~ internal_ipv4_address {
                      + subnet_id = "xxxxxxxxxxxxxxxxxx"
                        # (1 unchanged attribute hidden)
                    }
                }
            }

            # (1 unchanged block hidden)
        }
```

**Comment by apilikov** [comment]
Could you please open separate issues for each of the problems.

**Comment by miksir** [comment]
For fix issue from my last comment I posted PR #197

**Comment by apilikov** [comment]
Posted https://github.com/yandex-cloud/terraform-provider-yandex/issues/202 for that. Closing the issue.
